libraries/AP_HAL/I2CDevice.h

```diff
@@ -89,6 +89,28 @@ class I2CDeviceManager {
         // Not implemented
         return nullptr;
     }
+
+    /*
+      get mask of bus numbers for all configured I2C buses
+     */
+    virtual uint8_t get_i2c_bus_mask(void) const { return 0x0F; }
+
+    /*
+      get mask of bus numbers for all configured external I2C buses
+     */
+    virtual uint8_t get_i2c_bus_mask_external(void) const { return 0x0F; }
+
+    /*
+      get mask of bus numbers for all configured internal I2C buses
+     */
+    virtual uint8_t get_i2c_bus_mask_internal(void) const { return 0x01; }
```

Comment:
you are already in a I2CDeviceManager class... this could just be called "get_bus_mask_*()".

Reply:
Can we make this uint32_t? At least then it would be compatible with most Linux boards we support.  It's not compatible today anyway because the numbers can change... that's why I added support for traversing sysfs to get a bus by the controller name

Reply:
we loop over all values checking a mask, using a uint32_t would take too long
we could make it a list instead?

Reply:
right now it looks like all the linux boards use i2c bus numbers 4 or less. Is that likely to change?

Reply:
> we loop over all values checking a mask, using a uint32_t would take too long
> we could make it a list instead?

32 is still very small for a "do it on init only". If it's available on ARM, then __builtin_ffs would be an option to "return the next bit set"

> right now it looks like all the linux boards use i2c bus numbers 4 or less. Is that likely to change?

MS5611 on minlure is on bus 10

Reply:
yes, 32 is fine. I was being an idiot and thinking we'd loop over 2^32 :-)